src/compiler/parser.ts

```diff
@@ -2538,10 +2567,14 @@ namespace ts {
             return finishNode(node);
         }
 
-        function parseFunctionOrConstructorType(kind: SyntaxKind): FunctionOrConstructorTypeNode {
+        function parseFunctionOrConstructorType(kind: SyntaxKind): FunctionOrConstructorTypeNode | JSDocConstructorType {

```

Comment:
Could provide a more specific type to `kind`. I only see this being called with `kind` of `SyntaxKind.FunctionType` of `SyntaxKind.ConstructorType`, never `SyntaxKind.JSDocConstructorType`.